core/primitives/src/stateless_validation/state_witness.rs

```diff
@@ -187,17 +224,10 @@ impl ChunkStateWitness {
             signature_differentiator: "ChunkStateWitness".to_string(),
             _deprecated_new_transactions: vec![],
             _deprecated_new_transactions_validation_state: PartialState::default(),
-        }
-    }
-
-    pub fn chunk_production_key(&self) -> ChunkProductionKey {
-        ChunkProductionKey {
-            shard_id: self.chunk_header.shard_id(),
-            epoch_id: self.epoch_id,
-            height_created: self.chunk_header.height_created(),
-        }
+        })
     }
 
+    /// Mainly used for testing.
```

Comment:
```suggestion
    /// Used for testing.
```
nit to be clear on usecase